group/mod/int_test.go

```diff
@@ -98,3 +101,19 @@ func TestIntClone(t *testing.T) {
 		t.Error("Should not be equal")
 	}
 }
+
+type hh struct{}
+
+func (h *hh) Hash() hash.Hash {
+	return sha256.New()
+}
+
+func TestScalarHash(t *testing.T) {
+	token := make([]byte, 32)
+	rand.Read(token)
+	modulo := big.NewInt(65535)
+	var v int64 = 65500
+	i := new(Int).Init64(v, modulo)
+	_, err := i.Hash(new(hh), bytes.NewReader(token))
+	require.NoError(t, err)
+}
```

Comment:
Could we have also a test with an "expected" value, so that we can easily test compatibility with another implementation?

Reply:
Done